contracts/AMMWrapper.sol

```diff
@@ -241,8 +228,18 @@ contract AMMWrapper is IAMMWrapper, StrategyBase, ReentrancyGuard, BaseLibEIP712
                 weth.deposit{ value: msg.value }();
             }
         } else {
+            // Declare the 'userAddr sends takerAssetAddr to this contract' SpendWithPermit struct from _order parameter
+            SpenderLibEIP712.SpendWithPermit memory takerAssetPermit = SpenderLibEIP712.SpendWithPermit(
+                _order.takerAssetAddr,
+                address(this),
+                _order.userAddr,
+                address(this),
+                _order.takerAssetAmount,
+                AMMLibEIP712._getOrderHash(_order),
+                uint64(_order.deadline)
+            );
             // other ERC20 tokens
```

Comment:
I think this comment could be moved to line 231 and combine together to be `Transfer other ERC20 tokens to this contract`